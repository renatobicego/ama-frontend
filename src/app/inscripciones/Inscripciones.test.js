import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'
import Inscripciones from './page'

test('renders content', () => {
    const inscripciones = render(<Inscripciones />)

    expect(inscripciones).toMatchSnapshot()

    //Renders both divs to each inscripciones pages
    const federacion = screen.getByText('Federación Anual AMA')
    expect(federacion).toBeInTheDocument()
    const torneos = screen.getByText('Inscripción a Torneos')
    expect(torneos).toBeInTheDocument()

    //Both divs have href to page
    const links = screen.getAllByRole('link')
    expect(links[0].getAttribute('href')).toBe('/inscripciones/federaciones')
    expect(links[1].getAttribute('href')).toBe('/inscripciones/torneos')

    //Both images rendered
    const imgs = screen.getAllByAltText('Inscripciones')
    expect(imgs.length).toBe(2)
})