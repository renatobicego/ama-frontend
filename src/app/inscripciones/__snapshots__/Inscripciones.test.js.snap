// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="pt-[17vh] lg:pt-44 2xl:pt-52 pb-20"
      >
        <section
          class="size-section xl:mt-6"
        >
          <h2
            class="text-title title-section text-left"
          >
            Inscripciones
          </h2>
          <div
            class="items-center w-full md:w-[90%] flex justify-center gap-6 md:justify-between flex-wrap mt-10"
          >
            <a
              class="bg-tag-bg/50 relative w-4/5 md:w-[45%]"
              href="/inscripciones/federaciones"
              role="link"
            >
              <img
                alt="Inscripciones"
                class="object-cover object-center relative -z-10"
                data-nimg="1"
                decoding="async"
                height="400"
                loading="lazy"
                src="/_next/image?url=%2Fimgs%2Finscripciones1.png&w=1080&q=75"
                srcset="/_next/image?url=%2Fimgs%2Finscripciones1.png&w=640&q=75 1x, /_next/image?url=%2Fimgs%2Finscripciones1.png&w=1080&q=75 2x"
                style="color: transparent;"
                width="500"
              />
              <h3
                class="text-center font-title font-semibold text-base sm:text-lg md:text-xl lg:text-2xl w-52 md:w-96 text-white z-10 center-absolute"
              >
                Federaci贸n Anual AMA
              </h3>
            </a>
            <a
              class="bg-tag-bg/50 relative w-4/5 md:w-[45%]"
              href="/inscripciones/torneos"
              role="link"
            >
              <img
                alt="Inscripciones"
                class="object-cover object-center relative -z-10"
                data-nimg="1"
                decoding="async"
                height="400"
                loading="lazy"
                src="/_next/image?url=%2Fimgs%2Finscripciones2.png&w=1080&q=75"
                srcset="/_next/image?url=%2Fimgs%2Finscripciones2.png&w=640&q=75 1x, /_next/image?url=%2Fimgs%2Finscripciones2.png&w=1080&q=75 2x"
                style="color: transparent;"
                width="500"
              />
              <h3
                class="text-center font-title font-semibold text-base sm:text-lg md:text-xl lg:text-2xl w-52 md:w-96 text-white z-10 center-absolute"
              >
                Inscripci贸n a Torneos
              </h3>
            </a>
          </div>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="pt-[17vh] lg:pt-44 2xl:pt-52 pb-20"
    >
      <section
        class="size-section xl:mt-6"
      >
        <h2
          class="text-title title-section text-left"
        >
          Inscripciones
        </h2>
        <div
          class="items-center w-full md:w-[90%] flex justify-center gap-6 md:justify-between flex-wrap mt-10"
        >
          <a
            class="bg-tag-bg/50 relative w-4/5 md:w-[45%]"
            href="/inscripciones/federaciones"
            role="link"
          >
            <img
              alt="Inscripciones"
              class="object-cover object-center relative -z-10"
              data-nimg="1"
              decoding="async"
              height="400"
              loading="lazy"
              src="/_next/image?url=%2Fimgs%2Finscripciones1.png&w=1080&q=75"
              srcset="/_next/image?url=%2Fimgs%2Finscripciones1.png&w=640&q=75 1x, /_next/image?url=%2Fimgs%2Finscripciones1.png&w=1080&q=75 2x"
              style="color: transparent;"
              width="500"
            />
            <h3
              class="text-center font-title font-semibold text-base sm:text-lg md:text-xl lg:text-2xl w-52 md:w-96 text-white z-10 center-absolute"
            >
              Federaci贸n Anual AMA
            </h3>
          </a>
          <a
            class="bg-tag-bg/50 relative w-4/5 md:w-[45%]"
            href="/inscripciones/torneos"
            role="link"
          >
            <img
              alt="Inscripciones"
              class="object-cover object-center relative -z-10"
              data-nimg="1"
              decoding="async"
              height="400"
              loading="lazy"
              src="/_next/image?url=%2Fimgs%2Finscripciones2.png&w=1080&q=75"
              srcset="/_next/image?url=%2Fimgs%2Finscripciones2.png&w=640&q=75 1x, /_next/image?url=%2Fimgs%2Finscripciones2.png&w=1080&q=75 2x"
              style="color: transparent;"
              width="500"
            />
            <h3
              class="text-center font-title font-semibold text-base sm:text-lg md:text-xl lg:text-2xl w-52 md:w-96 text-white z-10 center-absolute"
            >
              Inscripci贸n a Torneos
            </h3>
          </a>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
